### PR TITLE
More SVG

### PR DESCRIPTION
**Pull request by jzwar**
# Overview
More svg

## Addressed issues
* Could not export parameter spline nicely

## Showcase
``` python
import splinepy as spp
from gustaf import Vertices as _Vertices

# Parameter spline
geometry = spp.helpme.create.box(1, 1)
geometry.elevate_degrees([0, 1])
parameter_spline = spp.Bezier(
    degrees=[2, 2],
    control_points=np.array(
        [0.05, 0.12, 0.22, 0.13, 0.2, 0.1, 0.2, 0.02, 0.14]
    ).reshape(-1, 1),
)
geometry.spline_data["field"] = parameter_spline
geometry.show_options["data"] = "field"
geometry.show_options["control_points"] = False
geometry.show_options["knots"] = True
geometry.show_options["knot_c"] = "black"
geometry.show_options["vmin"] = 0.05
geometry.show_options["vmax"] = 0.22
geometry.show_options["cmap"] = "jet"
vertices = _Vertices(geometry.cps)
vertices.vertex_data["para"] = parameter_spline.cps.ravel()
vertices.show_options["r"] = 0.04
vertices.show_options["vmin"] = 0.05
vertices.show_options["vmax"] = 0.22

spp.io.svg.export(
    "ParameterSpline.svg",
    geometry, vertices,
)
```

results in 
ParameterSpline

## Checklists
* [x] Documentations are up-to-date.
* [ ] Added example(s)
* [x] Added test(s)

### CODE REVIEW

File: splinepy/io/svg.py [hunk 1]
```diff
@@ -984,26 +1081,38 @@ def export(
...
+    if (len(objects_to_plot) == 1) and isinstance(objects_to_plot[0], list):
+        objects_to_plot = objects_to_plot[0]
+
+    for object in objects_to_plot:
+        if isinstance(object, Spline):
+            if not 1 <= object.para_dim <= 2:
+                raise ValueError("Only lines and surfaces supported")
+            if not object.dim == 2:
+                raise ValueError("Only 2D geometries supported")
+        elif isinstance(object, _GustafBase):
```
Comment (j042) [inline splinepy/io/svg.py:1093]:
```suggestion
        elif isinstance(object, _Vertices):
```
You can just check against vertices. All the other mesh types inherit from vertices

File: splinepy/io/svg.py [hunk 2]
```diff
@@ -14,6 +14,9 @@
     _color_map = ModuleImportRaiser(_error_message_vedo_import, err)
     _get_color = ModuleImportRaiser(_error_message_vedo_import, err)
 
+from gustaf import Vertices as _Vertices
+from gustaf._base import GustafBase as _GustafBase
```
Comment (j042) [inline splinepy/io/svg.py:18]:
```suggestion
```
see below

File: tests/test_svg.py [hunk 3]
```diff
@@ -257,6 +257,31 @@ def color_function(data, on):  # noqa ARG001
...
+        )
+
+        with tempfile.TemporaryDirectory() as tmpd:
+            tmpf = c.to_tmpf(tmpd)
+            c.splinepy.io.svg.export(
+                tmpf,
+                vertices,
+                indent=False,
+                box_margins=0.4,
+                background_c="black",
```
Comment (j042) [inline tests/test_svg.py:273]:
```suggestion
                background="black",
```
`_c` is `_c`onfusing

File: splinepy/io/svg.py [hunk 4]
```diff
@@ -984,26 +1081,38 @@ def export(
...
     # Determine bounding box of all elements
-    ctps_bounds = splines[0].control_point_bounds
-    for spline in splines[1::]:
-        ctps_bounds[0, :] = _np.minimum(
-            spline.control_point_bounds[0, :], ctps_bounds[0, :]
-        )
-        ctps_bounds[1, :] = _np.maximum(
-            spline.control_point_bounds[1, :], ctps_bounds[1, :]
-        )
+    ctps_bounds = _np.nan * _np.ones((2, 2))
```
Comment (j042) [inline splinepy/io/svg.py:1100]: Is this just to avoid initial init?

File: splinepy/io/svg.py [hunk 5]
```diff
@@ -984,26 +1081,38 @@ def export(
...
+    for object in objects_to_plot:
+        if isinstance(object, Spline):
+            ctps_bounds[0, :] = _np.fmin(
+                object.control_point_bounds[0, :], ctps_bounds[0, :]
+            )
+            ctps_bounds[1, :] = _np.fmax(
+                object.control_point_bounds[1, :], ctps_bounds[1, :]
```
Comment (j042) [inline splinepy/io/svg.py:1107]: this line will recompute

**Review by j042 — APPROVED** [review]
Great addition to smoothness, thank you!

File: splinepy/io/svg.py (hunk 5 above)
Comment (jzwar) [reply splinepy/io/svg.py:1107]: What do you mean?

File: tests/test_svg.py (hunk 3 above)
Comment (jzwar) [reply tests/test_svg.py:273]: Is similar to "knot_c", "control_point_c"... if you really want I will change this

File: splinepy/io/svg.py (hunk 4 above)
Comment (jzwar) [reply splinepy/io/svg.py:1100]: Yes, otherwise a check has to be performed what type of object we are dealing with

File: splinepy/io/svg.py [hunk 6]
```diff
@@ -984,26 +1081,38 @@ def export(
...
+            )
```
Comment (j042) [inline splinepy/io/svg.py:1108]:
```suggestion
            bounds = object.control_point_bounds
            ctps_bounds[0, :] = _np.fmin(
                bounds[0, :], ctps_bounds[0, :]
            )
            ctps_bounds[1, :] = _np.fmax(
                bounds[1, :], ctps_bounds[1, :]
            )
```
It recomputes bounds. Minor overhead, but it is an avoidable.
gus obj doesn't recompute.

File: tests/test_svg.py (hunk 3 above)
Comment (j042) [reply tests/test_svg.py:273]: I would like the change, unless there're other sub-options regarding background that I am missing. Good point though.

**Pull request merged**